src/charactersheet/services/character/status_service_components/magical_component.js

```diff
@@ -21,33 +23,84 @@ export function MagicalStatusServiceComponent() {
     var self = this;
 
     self.statusIdentifier = 'Status.Magical';
+    self.spellSlots = ko.observableArray([]);
 
-    self.init = function() {
-        Notifications.spellSlots.changed.add(self.dataHasChanged);
+    self.init = async function() {
+        await self.load();
+        self.setUpSubscriptions();
+    };
+
+    self.setUpSubscriptions = () => {
+        Notifications.spellslot.added.add(self.spellSlotAdded);
+        Notifications.spellslot.changed.add(self.spellSlotChanged);
+        Notifications.spellslot.deleted.add(self.spellSlotDeleted);
+        Notifications.coreManager.changing.add(self.clear);
+    };
+
+    self.load = async () => {
+        if (ko.utils.unwrapObservable(CoreManager.activeCore().type.name) !== 'character') {
+            return;
+        }
+        self.spellSlots([]);
+        const response = await SpellSlot.ps.list({coreUuid: CoreManager.activeCore().uuid()});
+        self.spellSlots(response.objects);
+        if (!self.spellSlots().length) {
```

Comment:
is this checking if the array is empty? could `self.spellSlots().length == 0` be used instead? it's more clear

Reply:
This also covers the case that length is undefined in addition to an empty list. The observableArray should never not have a length, so it is equivalent.